### PR TITLE
CHI-2836: Container for Private AI scheduled ECS task

### PR DESCRIPTION
**Pull request by stephenhand**
## Description

Working contain er for use in the Private AI integration. Has no terraformed infrastructure support yet, but has a manually created ECR repo, ECS task definition and SQS queue for sending test messages

- Container based on the PrivateAI base container with node support added and a custom entrypoint
- A node application that handles reading incoming scrub requests from SQS, loading the original transcript from s3, running it through Private AI and saving the scrubbed version back to s3 in a different location

Out of Scope:

Code in controller code to post a message to a 'completed' queue - this is to be done in a later task

### Checklist
- [ ] Corresponding issue has been opened
- [ ] New tests added
- [ ] Feature flags / configuration added

### Other Related Issues

None

### Verification steps

1. If needed build & push the container to `712893914485.dkr.ecr.us-east-1.amazonaws.com/development-transcript-scrubber-poc:latest` following AWS instructions - but the latest version is deployed
2. Post a message to the test queue here: https://us-east-1.console.aws.amazon.com/sqs/v3/home?region=us-east-1#/queues/https%3A%2F%2Fsqs.us-east-1.amazonaws.com%2F712893914485%2Fdevelopment-transcript-scrubber-pending-poc/send-receive - message format: `{ "bucket":"tl-aselo-docs-as-development", "key":"full/path/to/uscrubbed/transcript/with/no/starting/slash.json" }`
3. Check that a new file ghas been added to the same bucket with the same path except under the 'scrubbed-transcripts' folder rather than the 'transcripts' folder

### AFTER YOU MERGE

1. Cut a release tag using the GitHub workflow. Wait for it to complete and notify in the #aselo-deploys Slack channel.
5. Comment on the ticket with the release tag version AND any additional instructions required to configure an environment to test the changes.
6. Only then move the ticket into the QA column in JIRA

You are responsible for ensuring the above steps are completed. If you move a ticket into QA without advising what version to test, the QA team will assume the latest tag has the changes. If it does not, the following confusion is on you! :-P

### CODE REVIEW

File: hrm-domain/scheduled-tasks/transcript-scrubber/samples/sample1.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "text": "Some sample text about me, Mr Stephen James Hand of 110 Dangan Heights, Galway, who likes making soup and eating soup."
```
Comment (nickhurlburt) [inline hrm-domain/scheduled-tasks/transcript-scrubber/samples/sample1.json:2]: Might not want this in a public repo

File: hrm-domain/scheduled-tasks/transcript-scrubber/index.ts [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+export const executeTask = async () => {
+  const processingLatestFinishTime = Date.now() + MAX_PROCESSING_RUN_TIME_MILLIS;
+  await waitForPrivateAiToBeReady();
+  let processedMessages = 0;
+  while (await pollQueue()) {
+    processedMessages++;
+    if (Date.now() > processingLatestFinishTime) {
+      console.warn(
+        `Could not process all the pending messages in the configured window of ${Math.round(
```
Comment (nickhurlburt) [inline hrm-domain/scheduled-tasks/transcript-scrubber/index.ts:124]: I was looking for the monitoring for whether the configured window is right. This probably works for the time being, though it's manual and requires active attention. Ideally we want something more structured once this is in place... maybe a datadog metric or something.

**Comment by nickhurlburt** [comment]
Gave this a quick scan, not enough to feel I understand it enough for approval. Generally looks good. Going in, my questions were about monitoring and network architecture. Monitoring is mentioned in another comment. I'm curious about subnet / network permissions and such, but that may be a follow-on (and I don't recall if that was addressed already in the design doc).

File: hrm-domain/scheduled-tasks/transcript-scrubber/index.ts (hunk 2 above)
Comment (stephenhand) [reply hrm-domain/scheduled-tasks/transcript-scrubber/index.ts:124]: Monitoring is covered in the doc and has a separate ticket

**Review by murilovmachado — APPROVED** [review]
Looks good to me, Steve.

**Pull request merged**